From 1f4458eedd2a397ad967e77bec5c1fe1c7942e31 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 23 Apr 2019 12:23:12 +0300
Subject: [PATCH 170/386] arm64: dts: a7k: fix setup 7040db-E network
 configuration

Patch include 2 fixes:
1. Packet Processor port0 connected to SFP port
2. Packet Processor port1 to PHY reg0

Change-Id: I07f5f9cfee977bfcf92c6081e9ddabb824f18bbd
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/7974
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db-E.dts | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
index 7c6bed6975c6..9a214dc773a7 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-E.dts
@@ -20,6 +20,10 @@
 		ethernet0 = &cp0_eth0;
 		ethernet1 = &cp0_eth1;
 	};
+
+	sfp_eth0: sfp-eth0 {
+		compatible = "sff,sfp";
+	};
 };
 
 &uart0 {
@@ -76,9 +80,6 @@
 	phy0: ethernet-phy@0 {
 		reg = <0>;
 	};
-	phy1: ethernet-phy@2 {
-		reg = <1>;
-	};
 };
 
 &cp0_ethernet {
@@ -87,16 +88,16 @@
 
 &cp0_eth0 {
 	status = "okay";
-	/* Network PHY */
-	phy = <&phy0>;
 	phy-mode = "sgmii";
 	/* Generic PHY, providing serdes lanes */
 	phys = <&cp0_comphy2 0>;
+	managed = "in-band-status";
+	sfp = <&sfp_eth0>;
 };
 
 &cp0_eth1 {
 	status = "okay";
-	phy = <&phy1>;
+	phy = <&phy0>;
 	phy-mode = "rgmii-id";
 };
 
-- 
2.17.1

